pkg/pillar/cmd/baseosmgr/handlebaseos.go

```diff
@@ -1018,3 +1021,107 @@ func handleOtherPartRebootReason(ctxPtr *baseOsMgrContext, status *types.BaseOsS
 		status.SetError(reason, ctxPtr.rebootTime)
 	}
 }
+
+//isImageInErrorState returns true if we try to update to IMGB without success
+//also returns ZbootStatus
+func isImageInErrorState(ctxPtr *baseOsMgrContext) (bool, *types.ZbootStatus) {
+	otherPartName := zboot.GetOtherPartition()
+	partStatus := getZbootStatus(ctxPtr, otherPartName)
+	if partStatus == nil {
+		log.Functionf("No other partition status for %s",
+			otherPartName)
+		return false, nil
+	}
+	shortVerOtherPart := partStatus.ShortVersion
+	if shortVerOtherPart == "" {
+		log.Functionf("Other partition has no version")
+		return false, partStatus
+	}
+	if partStatus.PartitionState != "inprogress" {
```

Comment:
In makes sense to also check that the current image is in active state.

Reply:
Added